src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java

```diff
@@ -1,16 +1,47 @@
 // I'm bad at naming things. Please come up with a better name...
 package com.spartronics4915.frc2020.subsystems;
 
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.spartronics4915.frc2020.Constants;
 import com.spartronics4915.lib.subsystems.SpartronicsSubsystem;
 
+import 
+
+import edu.wpi.first.wpilibj.DigitalInput;
+
 public class PanelRotator extends SpartronicsSubsystem {
+    private final CANSparkMax mSpinMotor;
+    private final CANSparkMax mExtendMotor;
+    private final DigitalInput mBeamSensorUp;
+    private final DigitalInput mBeamSensorDown;
+    private final ColorSensorV3 mColorSensorRight;
+    private final ColorSensorV3 mColorSensorLeft;
+    private final I2C.Port i2cPort = 
 
     public PanelRotator() {
-        
+        mBeamSensorUp = new DigitalInput(Constants.kBeamSensorUpID);
+        mBeamSensorDown = new DigitalInput(Constants.kBeamSensorDownID);
+        mSpinMotor = new CANSparkMax(Constants.kSpinMotorID, MotorType.kBrushless);
+        mExtendMotor = new CANSparkMax(Constants.kExtendMotorID, MotorType.kBrushless);
+        mColorSensorLeft = new ColorSensorV3(Constants.kColorSensorLeftID);
```

Comment:
Okay, I kind of complicated things - totally forgot that there's only _one_ I2C port on the RoboRIO. We don't need to bother with constants or that I2C port variable above. You can just write [`new ColorSensorV3(I2C.Port.kOnboard)`](https://www.chiefdelphi.com/t/writing-code-for-a-color-sensor/167303/11).

That's kind of bad - the design team is planning for two color sensors. We'll need to tell them that there's only one port...

Reply:
^ No longer relevant